src/common/offline_transformations/include/mask_attribute.hpp

```diff
@@ -168,11 +182,14 @@ class Mask : public std::vector<std::set<uint64_t>>,
         m_need_initialization = false;
         // recursively apply callbacks for each dependent mask
         for (const auto & m_dependency : m_dependencies) {
+            if (m_dependency == mask.get())
+                continue;
             if (!m_dependency->apply_callback(shared_from_this())) {
                 return false;
             }
         }
-        return true;
+
+        return mask->apply_callback(shared_from_this());
```

Comment:
This is needed to define the order zero dimensions are propagating. Apply callback firstly visiting all their neighbors of node except "parent" node, which called current callback (children->apply_callback(parent)), and only then "parent"  callback is called. 
This logic is used in conv mask propagation, when the rout of mask should be following: (input->output->weights) but not (input->output->input)